platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1573,58 +1573,47 @@ private void assertUsageIsMutable() {
     }
 
     /**
-     * Conditionally warn when the usage of the configuration changes. In the following cases, the warning will not be emitted:
-     * <ol>
-     *     <li>Changes to the usage of the detached configurations should NOT warn (this done by the Kotlin plugin).</li>
-     *     <li>Configurations with a legacy role should NOT warn when changing usage, since the role-locked configuration APIs are still incubating</li>
-     * </ol>
+     * Conditionally warn when the usage of the configuration changes.
+     *
+     * <p>If this configuration was created with the legacy role, meaning it did not have a role set upon creation, this
+     * method will not emit a warning. We will want to remove this condition once the role-locked configuration creation
+     * APIs are de-incubated.</p>
+     *
+     * <p>If a usage is being disabled for a detached configuration, this method does not emit a warning.
+     * KMP disables consumable for detached configurations even though this is not necessary. This condition
+     * can be disabled with an internal feature flag system property for testing purposes.</p>
+     *
+     * <p>This method will also not warn if the current and new value are the same. We also want to eventually remove
+     * this condition, as build logic should not attempt to mutate a configuration's roles if that configuration
+     * has a role set upon creation. This condition can be disabled with an internal feature flag system property
```

Comment:
I thought about naming the flag here again, but I wanted to avoid the duplication of the string, and its super obvious what that string is just looking at the code. The reason for this doc is explaining _why_ we are special casing-these scenarios.

I can reword to better clarify that only a single flag controls both of these special cases

Reply:
I think having to read the body of a method to understand the javadoc is kind of backwards.  The name is already repeated throughout the tests, too.  Maybe a static constant would be a good idea, then you could link it in the javadoc and ®eference it in the tests?

Reply:
I've revised the javadoc and removed the mention of the flag. In practice there is already a ton of comments in the method body that explain whats happening, and I think whats going on here is pretty clear with all the comments and javadoc. Adding the flag in the javadoc body would just make things redundant IMO.